### PR TITLE
Add table azure_compute_virtual_machine_scale_set_vm closes #423

### LINKED ISSUE #423: Add table azure_compute_virtual_machine_scale_set_vm
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Terraform does not support the resource.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic info

```sql
select
  name,
  scale_set_name,
  instance_id,
  id,
  vm_id,
  region,
  resource_group
from
  azure_compute_virtual_machine_scale_set_vm;
```

```
+--------------------------------------+------------------------------------+-------------+---------------------------------------------------------------------------------------------------------------
| name                                 | scale_set_name                     | instance_id | id                                                                                                            
+--------------------------------------+------------------------------------+-------------+---------------------------------------------------------------------------------------------------------------
| test_0                               | test                               | 0           | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot_rg/providers/Microsoft.Compute/virtu
| turbot-test-20201223-create-update_3 | turbot-test-20201223-create-update | 3           | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/provider
| turbot-test-20201223-create-update_1 | turbot-test-20201223-create-update | 1           | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/provider
+--------------------------------------+------------------------------------+-------------+---------------------------------------------------------------------------------------------------------------
```

### List Standard tier scale set vms

```sql
select
  name,
  scale_set_name,
  id,
  sku_name,
  sku_tier
from
  azure_compute_virtual_machine_scale_set_vm
where
   sku_tier = 'Standard';
```

```
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------
| name                                 | scale_set_name                     | id                                                                                                                          
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------
| test_0                               | test                               | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot_rg/providers/Microsoft.Compute/virtualMachineScale
| turbot-test-20201223-create-update_3 | turbot-test-20201223-create-update | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/providers/Microsoft.Co
| turbot-test-20201223-create-update_1 | turbot-test-20201223-create-update | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/providers/Microsoft.Co
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------

```

### List all virtual machines under a specific scale set

```sql
select
  name,
  scale_set_name,
  id,
  sku_name,
  sku_tier
from
  azure_compute_virtual_machine_scale_set_vm
where 
  scale_set_name = 'turbot-test-20201223-create-update';
```

```
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------
| name                                 | scale_set_name                     | id                                                                                                                          
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------
| turbot-test-20201223-create-update_3 | turbot-test-20201223-create-update | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/providers/Microsoft.Co
| turbot-test-20201223-create-update_1 | turbot-test-20201223-create-update | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot-test-20201223-create-update/providers/Microsoft.Co
+--------------------------------------+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------

```

</details>

### CODE REVIEW

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+func listAzureComputeVirtualMachineScaleSetVms(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listAzureComputeVirtualMachineScaleSetVms")
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		return nil, err
+	}
+
+	scaleSet := h.Item.(compute.VirtualMachineScaleSet)
+	if scaleSet.Name == nil {
```
Comment (bigdatasourav) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:254]: this check is not required here

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+			},
+			{
+				Name:        "sku_tier",
+				Description: "Specifies the tier of virtual machines in a scale set VM.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Sku.Tier"),
+			},
+			{
+				Name:        "vm_id",
+				Description: "zure VM unique ID.",
```
Comment (bigdatasourav) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:105]:
```suggestion
				Description: "Azure VM unique ID.",
```

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("VirtualMachineScaleSetVMProperties.AdditionalCapabilities"),
+			},
+			{
+				Name:        "availability_set",
+				Description: "Specifies information about the availability set that the virtual machine should be assigned to. Virtual machines specified in the same availability set are allocated to different nodes to maximize availability. For more information about availability sets, see [Manage the availability of virtual machines](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-windows-manage-availability?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json). <br><br> For more information on Azure planned maintenance, see [Planned maintenance for virtual machines in Azure](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-windows-planned-maintenance?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json) <br><br> Currently, a VM can only be added to availability set at creation time. An existing VM cannot be added to an availability set.",
```
Comment (bigdatasourav) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:117]:
```suggestion
				Description: "Specifies information about the availability set that the virtual machine should be assigned to.",
```

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("VirtualMachineScaleSetVMProperties.StorageProfile"),
+			},
+			{
+				Name:        "zones",
+				Description: "The Logical zone list for scale set.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Azure standard columns
```
Comment (bigdatasourav) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:197]:
```suggestion
			// Steampipe standard columns
```

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+func listAzureComputeVirtualMachineScaleSetVms(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listAzureComputeVirtualMachineScaleSetVms")
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		return nil, err
+	}
+
+	scaleSet := h.Item.(compute.VirtualMachineScaleSet)
+	resourceGrooupName := strings.ToLower(strings.Split(*scaleSet.ID, "/")[4])
```
Comment (bigdatasourav) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:254]:
```suggestion
	resourceGroupName := strings.ToLower(strings.Split(*scaleSet.ID, "/")[4])
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceGroupNotFound", "ResourceNotFound", "404"}),
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listAzureComputeVirtualMachineScaleSets,
+			Hydrate:       listAzureComputeVirtualMachineScaleSetVms,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "Name of the scale set VM.",
```
Comment (misraved) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:31]: Should this be "Name of the virtual machine scale set"?

File: azure/table_azure_compute_virtual_machine_scale_set_vm.go [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("VirtualMachineScaleSetVMProperties.AdditionalCapabilities"),
+			},
+			{
+				Name:        "availability_set",
+				Description: "Specifies information about the availability set that the virtual machine should be assigned to. Virtual machines specified in the same availability set are allocated to different nodes to maximize availability. For more information about availability sets, see [Manage the availability of virtual machines](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-windows-manage-availability?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json). <br><br> For more information on Azure planned maintenance, see [Planned maintenance for virtual machines in Azure](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-windows-planned-maintenance?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json) <br><br> Currently, a VM can only be added to availability set at creation time. An existing VM cannot be added to an availability set.",
```
Comment (misraved) [inline azure/table_azure_compute_virtual_machine_scale_set_vm.go:117]: Can we remove the line breaks and make the description slightly concise?

File: docs/tables/azure_compute_virtual_machine_scale_set_vm.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  instance_id,
+  id,
+  vm_id,
+  region,
+  resource_group
+from
+  azure_compute_virtual_machine_scale_set_vm;
+```
+
+### List Standard tier scale set vms
```
Comment (misraved) [inline docs/tables/azure_compute_virtual_machine_scale_set_vm.md:22]:
```suggestion
### List Standard tier scale set virtual machines
```

File: docs/tables/azure_compute_virtual_machine_scale_set_vm.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  id,
+  sku_name,
+  sku_tier
+from
+  azure_compute_virtual_machine_scale_set_vm
+where 
+  scale_set_name = 'my_vm_scale';
+```
+
+### View Network Security Group Rules for a VM
```
Comment (misraved) [inline docs/tables/azure_compute_virtual_machine_scale_set_vm.md:52]:
```suggestion
### View Network Security Group Rules for a virtual machine
```

**Review by misraved — COMMENTED** [review]
Please review if the VM acronym used in the descriptions can be updated to virtual machines.
Make changes wherever necessary

**Pull request merged**